frontend/src/features/public-form/PublicFormProvider.tsx

```diff
@@ -147,27 +145,9 @@ export const PublicFormProvider = ({
   )
 
   useEffect(() => {
-    if (data) {
-      if (!cachedDto) {
-        trackVisitPublicForm(data.form)
-        setCachedDto(data)
-      } else if (!desyncToastIdRef.current && !isEqual(data, cachedDto)) {
-        desyncToastIdRef.current = toast({
-          status: 'warning',
-          title: (
-            <Text textStyle="subhead-1">
-              The form has been modified and your submission may fail.
-            </Text>
-          ),
-          description: (
-            <Text as="span">
-              <Link href={window.location.href}>Refresh</Link> for the latest
-              version of the form.
-            </Text>
-          ),
-          duration: null,
-        })
-      }
+    if (data && !cachedDto) {
+      trackVisitPublicForm(data.form)
+      setCachedDto(data)
```

Comment:
if we do not want to track changes, we might not even need to cache anything and just need to set staleTime=Infinity in the query.

Reply:
oh yes, that's true. Did we decide to dynamically update preview form also? or just static as well? should we remove the cachedDto for preview forms too?

Reply:
yup, lets remove it for preview too until we can revisit it.

Reply:
[90904e8](https://github.com/opengovsg/FormSG/pull/4829/commits/90904e822d5db8485d7585a35227795b309648d4)